src/base/Identifiers.mli

```diff
@@ -0,0 +1,38 @@
+(*
+  This file is part of scilla.
+
+  Copyright (c) 2020 - present Zilliqa Research Pvt. Ltd.
+  
+  scilla is free software: you can redistribute it and/or modify it under the
+  terms of the GNU General Public License as published by the Free Software
+  Foundation, either version 3 of the License, or (at your option) any later
+  version.
+ 
+  scilla is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+  A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ 
+  You should have received a copy of the GNU General Public License along with
+  scilla.  If not, see <http://www.gnu.org/licenses/>.
+*)
+
+open ErrorUtils
+
+type 'rep ident = Ident of string * 'rep [@@deriving sexp]
```

Comment:
I've been meaning to do a refactoring like this one. We could try making the `ident` type opaque to make this more robust. I guess that will require providing functions to hash identifiers to be able to use them with hashmaps.